File: data-structures/hash-tables/python/hash_function_integer.py
# Raspberry Pi Foundation
# Developed to be used alongside Isaac Computer Science,
# part of the National Centre for Computing Education
# Usage licensed under CC BY-SA 4


def hash_integer(hash_key, number_of_slots):
    """Produce a hash value from an integer"""

    # Generate the hash value using the modulo operator
    hash_value = hash_key % number_of_slots

    # Return the hash value
    return hash_value


def main():
    """Test the hashing function"""

    # Generate a hash value from an integer hash key
    key = 12345
    slots = 97
    value = hash_integer(key, slots)
    print(f"Hash value for {key} is {value}")


# This code will run if this file is executed directly
# (i.e. not called by another program)
if __name__ == '__main__':
    main()

File: data-structures/hash-tables/c-sharp/hash_function_integer.cs
/*
Isaac Computer Science
Usage licensed under the Open Government Licence v3.0

Note: This file is designed to be copied out and compiled on your machine.
In order for it to compile properly you need to ensure that the project name is the same as the "namespace" in this file.
To run this file you need to:
1. Copy the contents
2. Paste them into the C# IDE of your choice (Visual Studio, for example)
3. Change the namespace to match your project (if necessary)
4. Compile the program
5. Run the program
*/

using System;

namespace IsaacCodeSamples
{
    class HashTables
    {
        // The Main method is the entry point for all C# programs
        public static void Main()
        {
            // Generate a hash value from an integer hash key
            int key = 12345;
            int slots = 97;
            int value = HashInteger(key, slots);
            Console.WriteLine($"Hash value for {key} is {value}");
        }

        // Produce a hash value from an integer
        public static int HashInteger(int hashKey, int numberOfSlots)
        {
            // Generate the hash value using the modulo operator
            int hashValue = hashKey % numberOfSlots;

            // Return the hash value
            return hashValue;
        }


    }
}